### PR TITLE
Fix accounts engine

### PR DESCRIPTION
**Pull request by sauloperez**
I'm adding and fixing the tests of the accounts engine

### CODE REVIEW

File: engines/account/app/services/account/signup_service.rb [hunk 1]
```diff
@@ -23,7 +20,7 @@ def create!(email)
     # @return [Account::User]
     def complete!(signup, options)
       ::ActiveRecord::Base.transaction do
-        user = ::Account::User.create(
+        user = ::Account::User.create!(
           email: signup.email,
           username: options[:username],
           password: options[:password],
```
Comment (enricostano) [inline engines/account/app/services/account/signup_service.rb]: @andresgutgon could you please paste here a diff with what we found out about the `SignupService`?

Comment (andresgutgon) [inline engines/account/app/services/account/signup_service.rb]: The thing is that adfing the `!` makes transaction fail when validations fail. But we're not resque it in the controller en this is failing bad in the front-end.

What I did is this:

``` ruby
signup.destroy if user.persisted?
```

Comment (sauloperez) [inline engines/account/app/services/account/signup_service.rb]: What we need to do instead is handle it in the controller. This is how transaction block are supposed to work. What you suggest is implementing transactions by hand.

Comment (enricostano) [inline engines/account/app/services/account/signup_service.rb]: Another approach is to rescue the rollback inside the service, without leaking too much information to the controller.

The main goal for now is to make sure that we destroy the signup only if the user has been created.

We'll implement the rescue inside the `SignupService` but not sure if we have time now.

**Pull request merged**